client/src/main/java/io/pravega/client/tables/impl/TableSegmentEntry.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables.impl;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Entry in a {@link TableSegment}.
+ */
+@Getter
+@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
+public class TableSegmentEntry {
```

Comment:
Is this supposed to be public or package protected? Do I understand right that `TableSegment` isn't a concept we are exposing, that it is an internal implementation detail?

Reply:
Because the Controller **already** uses these classes I have no option but to make them public,

Reply:
Let's create an issue ahead of time so that we don't forget, please.

Reply:
#4647 and assigned to @shrids .